tools/integration_tests/rapid_appends/appends_test.go

```diff
@@ -34,353 +33,252 @@ const (
-			defer t.unmountPrimaryMount()
-
-			log.Printf("Running tests with flags: %v", flags)
-
-			// Initially create an unfinalized object.
-			t.createUnfinalizedObject()
-			defer t.deleteUnfinalizedObject()
-
-			// Initiate an append session using the primary file handle opened in append mode.
-			appendFileHandle := operations.OpenFileInMode(t.T(), path.Join(t.primaryMount.testDirPath, t.fileName), os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
-			_, err = appendFileHandle.WriteString(initialContent)
-			require.NoError(t.T(), err)
-
-			// Open a new file handle from the secondary mount to the same file.
-			newAppendFileHandle := operations.OpenFileInMode(t.T(), path.Join(t.secondaryMount.testDirPath, t.fileName), os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
-			defer operations.CloseFileShouldNotThrowError(t.T(), newAppendFileHandle)
-
-			// Attempt to append using the newly opened file handle.
-			// This append should succeed, confirming the takeover.
-			_, err = newAppendFileHandle.WriteString(appendContent)
-			assert.NoError(t.T(), err)
-
-			// Attempt to append using the original file handle.
-			// This should now fail, as its append session has been invalidated by the takeover.
-			_, _ = appendFileHandle.WriteString(appendContent)
-			err = appendFileHandle.Sync()
-			operations.ValidateESTALEError(t.T(), err)
-
-			// Syncing from the newly created file handle must succeed since it holds the active
-			// append session.
-			err = newAppendFileHandle.Sync()
-			assert.NoError(t.T(), err)
-
-			// Read from primary mount to validate the contents which has persisted in GCS after
-			// takeover from the secondary mount.
-			// Close the open append handle before issuing read on the file as Sync() triggered on
-			// ReadFile() due to BWH still being initialized, is expected to error out with stale NFS file handle.
-			operations.CloseFileShouldThrowError(t.T(), appendFileHandle)
-			expectedContent := t.fileContent + appendContent
-			content, err := operations.ReadFile(path.Join(t.primaryMount.testDirPath, t.fileName))
-			require.NoError(t.T(), err)
-			assert.Equal(t.T(), expectedContent, string(content))
-		}()
-	}
+
+	s.createUnfinalizedObject()
+	defer s.deleteUnfinalizedObject()
+
+	// Initiate an append session using the primary file handle.
+	appendFileHandle := operations.OpenFileInMode(s.T(), path.Join(s.primaryMount.testDirPath, s.fileName), os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
+	_, err := appendFileHandle.WriteString(initialContent)
+	require.NoError(s.T(), err)
+
+	// Open a new file handle from the secondary mount to the same file.
+	newAppendFileHandle := operations.OpenFileInMode(s.T(), path.Join(s.secondaryMount.testDirPath, s.fileName), os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
+	defer operations.CloseFileShouldNotThrowError(s.T(), newAppendFileHandle)
+
+	// This append should succeed, confirming the takeover.
+	_, err = newAppendFileHandle.WriteString(appendContent)
+	assert.NoError(s.T(), err)
```

Comment:
There is no point proceeding if this fails.

```suggestion
	require.NoError(s.T(), err)
```